envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -64,9 +70,19 @@ class EnvoyClustersFactory(
         return insecureClusters.map { cluster ->
             val upstreamTlsContext = createTlsContextWithSdsSecretConfig(cluster.name)
             val secureCluster = Cluster.newBuilder(cluster)
-            secureCluster.setTransportSocket(TransportSocket.newBuilder()
-                    .setName("envoy.transport_sockets.tls")
-                    .setTypedConfig(Any.pack(upstreamTlsContext)))
+
+            val matchTlsContext = Cluster.TransportSocketMatch.newBuilder()
+                    .setName(tlsProperties.tlsTransportSocketMatchName)
+                    .setMatch(
+                            Struct.newBuilder().putFields(tlsProperties.tlsContextMetadataMatchKey,
+                            Value.newBuilder().setBoolValue(true).build())
```

Comment:
The match can be a precalculated field.
Also, I think there is some indendation problem here.